set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -1,557 +1,600 @@
+		Date now = new Date();
+		userSet.setModified(now);
+	    }
+	    removeItemDuplicates(userSet);
+	    return userSet;
+	} catch (UserSetAttributeInstantiationException e) {
+	    throw new RequestBodyValidationException(I18nConstants.USERSET_CANT_PARSE_BODY,
+		    new String[] { e.getMessage() }, e);
+	} catch (JsonParseException e) {
+	    throw new UserSetInstantiationException("Json formating exception! " + e.getMessage(), e);
+	} catch (IOException e) {
+	    throw new UserSetInstantiationException("Json reading exception! " + e.getMessage(), e);
+	}
+    }
+
+    /**
+     * This method normalizes item list if they exist to remove duplicated items.
+     * 
+     * @param userSet
+     * @throws ParamValidationException
+     */
+    public void removeItemDuplicates(UserSet userSet) {
+	if (userSet.getItems() != null && !userSet.getItems().isEmpty()) {
+	    List<String> distinctItems = userSet.getItems().stream().distinct().collect(Collectors.toList());
+	    userSet.setItems(distinctItems);
+	}
+    }
+
+    public void validateWebUserSet(UserSet webUserSet) throws RequestBodyValidationException {
+
+	// validate title
+	if (webUserSet.getTitle() == null) {
+	    throw new RequestBodyValidationException(I18nConstants.USERSET_VALIDATION_MANDATORY_PROPERTY,
+		    new String[] { WebUserSetModelFields.TITLE });
+	}
+
+	// validate context
+	if (webUserSet.getContext() != null
+		&& !WebUserSetModelFields.VALUE_CONTEXT_EUROPEANA_COLLECTION.equals(webUserSet.getContext())) {
+	    throw new RequestBodyValidationException(I18nConstants.USERSET_VALIDATION_PROPERTY_VALUE,
+		    new String[] { WebUserSetModelFields.AT_CONTEXT, webUserSet.getContext() });
+	}
+
+	//validate visibility
+	if (webUserSet.getVisibility()!= null && !VisibilityTypes.isValid(webUserSet.getVisibility())) {
+	    throw new RequestBodyValidationException(I18nConstants.USERSET_VALIDATION_PROPERTY_VALUE,
+		    new String[] { WebUserSetModelFields.VISIBILITY, webUserSet.getVisibility()});
+	}
+	
+	// validate isDefinedBy and items - we should not have both of them
+	if (webUserSet.getItems() != null && webUserSet.getIsDefinedBy() != null) {
+	    throw new RequestBodyValidationException(I18nConstants.USERSET_VALIDATION_PROPERTY_NOT_ALLOWED,
+		    new String[] { WebUserSetModelFields.ITEMS, WebUserSetModelFields.TYPE_OPEN });
+	}
+    }
+
+    /* (non-Javadoc)
+     * @see eu.europeana.set.web.service.UserSetService#validateFavoriteUserSet(eu.europeana.set.definitions.model.UserSet, eu.europeana.set.definitions.model.UserSet)
+     */
+    public void validateFavoriteUserSet(UserSet webUserSet, UserSet storedUserSet) throws RequestBodyValidationException, UserSetNotFoundException {
+
+	// verify if the "type" is set to "BookmarkFolder", if true:
+	if (webUserSet.getType() != null && WebUserSetModelFields.DEFAULT_FAVORITE_TYPE.equals(webUserSet.getType())) {
+	    // check if set is open set, if true return HTTP 400;
+	    if (webUserSet.isOpenSet()) {
```

Comment:
please remove the rest of the code from this method and integrate the call in the validateWebUserSet. A bookmark folder is also always a closed set

Reply:
Removed the rest of the code from the method and integrated in validateWebUserSet